internal_ws/yksg/src/lib.rs

```diff
@@ -383,16 +383,26 @@ impl StopgapInterpreter {
                 expected,
                 target_bb,
             } => {
-                let b = self.read_int(cond) == 1;
-                if b != *expected {
-                    todo!() // FIXME raise error
+                if self.read_bool(cond) != *expected {
+                    todo!();
                 }
                 self.frames.last_mut().unwrap().bbidx = *target_bb;
             }
             t => todo!("{}", t),
         }
     }
 
+    fn read_bool(&self, src: &IRPlace) -> bool {
+        if let IRPlace::Const { val: Constant::Int(ConstantInt::UnsignedInt(UnsignedInt::U8(v))), ty: _ } = src {
+            *v != 0
+        } else if let TyKind::Bool = SIR.ty(&src.ty()).kind {
+            let ptr = self.locals().irplace_to_ptr(src);
+            unsafe { std::ptr::read::<u8>(ptr) != 0 }
```

Comment:
We could change this to `unsafe { std::ptr::read::<bool>(ptr as *const bool) != 0 }` to address your concerns about assuming `bool`s will always be `u8`.

Reply:
Yes, I've got that in a local branch, though the assumption is embedded in several other places. However, I still end up reading bytes whose value is > 1 (i.e. we've caused undefined behaviour somewhere).